### DISCUSSION TITLE
next/previous link equivalent in postgrest

### DISCUSSION THREAD

**Discussion by rv2931** [PostgREST/postgrest#3772] [Q&A] [answered]
Hello
I'm not an http specialist and I understand the choice of PostgRest project to conform to http standard and choose to dedicate body data for data structure.
But is there any solution that could supply the next/previous link solution to scrawl over multiple page results ?
This should be http compliant but is there any http semantic field that could be used for that ?
Next/previous link in API is really convenient to retrieve all data when needed by data visualization tool that are generally not really smart.
What is your opinion on that ?
Thank you for reading 
RV

**Comment by wolfgangwalther** [comment]
I guess you could try to write your own [media type handler](https://docs.postgrest.org/en/v12/references/api/media_type_handlers.html) for `application/json` (or a variant), which wraps the regular response in a "container". This container could have metadata, e.g. your links.

Not sure how far you'd come, but I would be interested to hear about that!

**Reply by rv2931** [reply]
I was not thinking to create a new format or something that was not standard. My thinking was more to find or divert somewhat an existing semantic http Header field like having named locations or next range, next resources 
I took a quick look to RFC but I didn't find something satisfying but maybe there are some experts in http that could find something

**Comment by rv2931** [comment, marked as answer]
@wolfgangwalther what about this web linking RFC ?
https://datatracker.ietf.org/doc/html/rfc8288

Itseems to allow Link headers for next/previous/start/end links
```
Link: <http://......>, rel="next"
Link: <http://......>, tel="previous"
```

I don't know it include specifying headers attributes to the link like
`Link: <http://....>, rel=next, range=items:101-200;`

As the RFC seems to define a closed list of link-params but in the other hand it shouldn't be a problem to add custom parameter like "range" to the list
Or this could make the need to manage query parameters limit/offset to maintain full compliance with RFC and process these query params as range header param as header param can't be defined in link header

**Reply by wolfgangwalther** [reply]
Sounds interesting. Could you create an issue along the lines of "Support Link headers etc..." instead of a discussion so we can track this as a feature request?
